File: ambeth/Ambeth.IoC/ambeth/ioc/factory/OrderState.cs
using System.Collections.Generic;

namespace De.Osthus.Ambeth.Ioc.Factory
{
    public class OrderState : List<BeanConfigState>
    {
        private int processedIndex = -1;

        public BeanConfigState ConsumeBeanConfigState()
        {
            if (processedIndex + 1 < Count)
            {
                return this[++processedIndex];
            }
            return null;
        }
    }
}

File: jambeth/jAmbeth.IoC/src/de/osthus/ambeth/ioc/factory/OrderState.java
package de.osthus.ambeth.ioc.factory;

import de.osthus.ambeth.collections.ArrayList;

public class OrderState extends ArrayList<BeanConfigState>
{
	private int processedIndex = -1;

	public BeanConfigState consumeBeanConfigState()
	{
		if (processedIndex + 1 < size())
		{
			return get(++processedIndex);
		}
		return null;
	}
}
